### DISCUSSION TITLE
Distributed Field Transfer

### DISCUSSION THREAD

**Discussion by termi-official** [mfem/mfem#2092] [Q&A] [answered]
Hi together,
is there an example, or some other implementation similar to the "Field Interpolation Miniapp", which works for distributed simulations? I.e. given two ParMeshes and a ParGridFunction given one of these, transfer the ParGridFunction onto the other ParMesh. The 'naive' way would be to transfer the process-local grid between all combinations of processes, but this seems to be really bad I guess this completely destroys any scaling.

If there is no implementation for this use case in MFEM yet, is someone aware of literature on this topic?

Thanks in advance,
Dennis

**Comment by vladotomov** [comment, marked as answer]
I think it can be done, we just haven't tried implementing it. The main ingredient (gslib) works in parallel, so you should be able to create a list of physical points (the FE nodes on the `mesh_1`) that you want to interpolate on every task, and then query the ParGridFunction (that lives on `mesh_2`), similar to `miniapps/pfindpts.cpp`. But I might be missing something.

**Reply by termi-official** [reply]
Thanks for the reply.
 I also realized in the meantime that the gslib wrapper gives additional information about the MPI rank to which the found point belongs, which is great! I also think that interpolate does already everything which I need by I had no time to try it yet. I.e. going through the interpolation code it should be something as simple as
```cpp
ParMesh* mesh1 = load_mesh1();
ParMesh* mesh2 = load_mesh2();
ParGridFunction pgf1 = load_gf1(mesh1);
ParGridFunction pgf2 = create_gf2(mesh2);
Vector vxyz = compute_nodes(pgf1);

FindPointsGSLIB finder(MPI_COMM_WORLD);
finder.Setup(mesh2);
finder.FindPoints(vxyz);
finder.Interpolate(pgf1, pgf2);

// now pgf2 has the correct values on each process.
```
since ``Interpolate`` seem to handle the inter-process communication for me, or do I miss something?

**Reply by vladotomov** [reply]
Yes, `Interpolate` takes care of the communication when the position is found on a different task.
